File: 2526B/apps mock/nineknights_Amanda.py
'''
OpenKattis
nineknights
Code by: Amanda Lim
'''

dx = [-1,-1,1,1,-2,-2,2,2]
dy = [-2,2,-2,2,-1,1,-1,1]

grid = [] # could be []*5
for i in range(5):
    str = input() # could be grid[i] = input()
    grid.append(str)

valid = True
knights = 0
for i in range(5):
    for j in range(5):
        if (grid[i][j] == '.'):
            continue
        knights += 1

        for k in range(8):
            if (((i + dx[k]) >= 0) and ((i + dx[k]) < 5) and ((j + dy[k]) >= 0) and ((j + dy[k]) < 5)):
                if (grid[i + dx[k]][j + dy[k]] == 'k'):
                    valid = False

if (valid and (knights == 9)):
    print("valid")
else:
    print("invalid")

File: 2526B/apps mock/nineknights_Amanda.cpp
/*
*OpenKattis
*nineknights
*Code by: Amanda Lim
*/

#include <bits/stdc++.h>
using namespace std;

int main ()
{
    ios::sync_with_stdio(0);
    cin.tie(0);
    cout.tie(0);

    int dx[] = {-1,-1,1,1,-2,-2,2,2};
    int dy[] = {-2,2,-2,2,-1,1,-1,1};

    char grid[5][5];
    for (int i = 0; i < 5; i++)
        for (int j = 0; j < 5; j++)
            cin >> grid[i][j];
    
    bool valid = true;
    int knights = 0;
    for (int i = 0; (i < 5) && valid; i++)
        for (int j = 0; (j < 5) && valid; j++) {
            if (grid[i][j] == '.')
                continue;

            knights++;

            for (int k = 0; (k < 8) && valid; k++) {
                if (((i + dx[k]) >= 0) && ((i + dx[k]) < 5) && ((j + dy[k]) >= 0) && ((j + dy[k]) < 5))
                    if (grid[i + dx[k]][j + dy[k]] == 'k')
                        valid = false;
            }
        }

    if (valid && (knights == 9))
        cout << "valid" << endl;
    else
        cout << "invalid" << endl;
    
    return 0;
}